stl/inc/mutex

```diff
@@ -34,24 +34,22 @@ _EXPORT_STD class condition_variable_any;
 struct _Mtx_internal_imp_mirror {
 #ifdef _CRT_WINDOWS
 #ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 8;
+    static constexpr size_t _Critical_section_size = 16;
 #else // _WIN64
-    static constexpr size_t _Critical_section_size = 4;
+    static constexpr size_t _Critical_section_size = 8;
 #endif // _WIN64
 #else // _CRT_WINDOWS
 #ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 56;
+    static constexpr size_t _Critical_section_size = 64;
 #else // _WIN64
-    static constexpr size_t _Critical_section_size = 32;
+    static constexpr size_t _Critical_section_size = 36;
 #endif // _WIN64
 #endif // _CRT_WINDOWS

```

Comment:
I suspected and then confirmed that this fails to compile for our non-Kernel32 builds (currently MSVC-internal only, sorry):
```
   101>D:\msvc\src\vctools\crt\github\stl\src\mutex.cpp(55,59): error C2607: static assertion failed [D:\msvc\src\vctools\crt\github\stl\msbuild\stl_base\xmt0\libcpmt_onecore\libcpmt.nativeproj]
             std::_Mtx_internal_imp_mirror::_Critical_section_size == Concurrency::details::stl_critical_section_max_size);
```

This is because `_STL_CONCRT_SUPPORT` is defined for Kernel32 only (this happens in MSVC-internal files):

https://github.com/microsoft/STL/blob/c8d1efb6d504f6392acf8f6d01fd703f7c8826c0/stl/src/primitives.hpp#L123-L132

I believe we were getting away with it earlier due to the less-than-or-equal checks, but your exactly-equal check notices the difference. (This may actually have meant that our recently introduced read of `_Count` directly from the representation wasn't working for non-Kernel32, and nobody noticed.)

Reply:
How do I detect non-Kernel32 builds? It seems that `_STL_CONCRT_SUPPORT` is defined only when building the library, and so `<mutex>` can't use it.

https://github.com/microsoft/STL/blob/c8d1efb6d504f6392acf8f6d01fd703f7c8826c0/stl/CMakeLists.txt#L497

Reply:
So I created a UWP app and verified that the read of `_Count` indeed gets the wrong value.

    alignas(std::mutex) unsigned char buf[sizeof(std::mutex)];
    int val = INT_MAX;
    std::memcpy(buf + sizeof(std::mutex) - sizeof val, &val, sizeof val);

    auto p = new(buf) std::mutex;
    try {
        p->lock();
    } catch (std::system_error&) {
        assert(false); // asserts on UWP
    }